src/absil/ilwrite.fs

```diff
@@ -4025,17 +4037,18 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
           let entrypointCodeChunk,next = chunk 0x06 next
           let globalpointerCodeChunk,next = chunk (if isItanium then 0x8 else 0x0) next
           
-          let debugDirectoryChunk,next = chunk (if pdbfile = None then 0x0 else sizeof_IMAGE_DEBUG_DIRECTORY) next
+          let debugDirectoryChunk,next = chunk (if pdbfile = false then 0x0 else sizeof_IMAGE_DEBUG_DIRECTORY) next
           // The debug data is given to us by the PDB writer and appears to 
           // typically be the type of the data plus the PDB file name.  We fill 
-          // this in after we've written the binary. We approximate the size according 
+          // this in after we've written the binary. We approximate the size to MAXPATH according 
           // to what PDB writers seem to require and leave extra space just in case... 
           let debugDataJustInCase = 40
+          let MAXPATH = 260
           let debugDataChunk,next = 
               chunk (align 0x4 (match pdbfile with 
-                                | None -> 0x0 
-                                | Some f -> (24 
-                                            + System.Text.Encoding.Unicode.GetByteCount(f) // See bug 748444
+                                | false -> 0x0 
+                                | true -> (24 
+                                            + MAXPATH
```

Comment:
To get into Microsoft/visualfsharp you'll need to have written the code so that every point we can see that it's 100% "safe" - i.e. there is absolutely zero risk of behaviour change.  So for this kind of thing you will need need to take the max between MAXPATH and GetByteCount, or leave the behaviour exactly the same as before when using a on-disk file as output.